src/common/reply.h

```diff
@@ -0,0 +1,13 @@
+#ifndef REPLY_H
+#define REPLY_H
+
+// Meta definitions
+#define RPL_META_MESSAGE(servername, numeric, message) (std::string(":") + servername + " " + numeric + " " + message + "\r\n")
+
+// Numeric replies in order
+#define RPL_WELCOME_001(servername, nick, user, host) (RPL_META_MESSAGE(servername, "001", "Welcome to the Internet Relay Network " + nick + "!" + user "@" + host))
```

Comment:
Here the IRCv3 spec calls for the message contents be prepended with the colon ```:```, while RFC 2812 does not have the colon.
Another difference is that in IRCv3, the first parameter should be the client nickname while RFC 2812 does not include the client nickname in the message contents.

| spec. | message parameters |
| --- | --- |
| RFC 2812 | ```Welcome to the Internet Relay Network <nick>!<user>@<host>``` |
| IRCv3 | ```<client> :Welcome to the <networkname> Network, <nick>[!<user>@<host>]``` |